compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1399,4 +1597,31 @@ private void defineInitFunctionParam(BLangSimpleVariable varNode) {
         varNode.symbol = paramSymbol;
         return;
     }
+
+    /**
+     * Used to store location data for encountered unknown types in `checkErrors` method.
+     *
+     * @since 0.985.0
+     */
+    class LocationData {
```

Comment:
check whether we can remove this and improve type def logic to print the error properly

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/11394 to track this.